src/uu/cp/src/cp.rs

```diff
@@ -1728,15 +1741,53 @@ fn copy_file(
         let mut permissions = source_metadata.permissions();
         #[cfg(unix)]
         {
-            use uucore::mode::get_umask;
-
             let mut mode = permissions.mode();
 
-            // remove sticky bit, suid and gid bit
-            const SPECIAL_PERMS_MASK: u32 = 0o7000;
-            mode &= !SPECIAL_PERMS_MASK;
+            let (is_preserve_mode, is_explicit_no_preserve_mode) = options.preserve_mode();
+            if !is_preserve_mode {
```

Comment:
maybe move these info a function ?

Reply:
did you mean the lines between 1749 to 1789?

Reply:
yes :)